### PR TITLE
Cgmes import update. Import EQ and SSH separately. TieLines

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**Does this PR introduce a new Powsybl Action implying to be implemented in simulators or pypowsybl?**
- [ ] Yes, the corresponding issue is [here](link)
- [x] No

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        <cim:IdentifiedObject.name>ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit1</cim:IdentifiedObject.name>
+        <cim:ActivePowerLimit.normalValue>91</cim:ActivePowerLimit.normalValue>
+        <cim:OperationalLimit.OperationalLimitSet rdf:resource="#_ACLineSegment-2-T1-OperationalLimitSet"/>
+        <cim:OperationalLimit.OperationalLimitType rdf:resource="#_ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit1-Patl"/>
+    </cim:ActivePowerLimit>
+    <cim:OperationalLimitType rdf:ID="_ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit1-Patl">
+        <cim:IdentifiedObject.name>ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit1-Patl</cim:IdentifiedObject.name>
+        <cim:OperationalLimitType.direction rdf:resource="http://iec.ch/TC57/CIM100#OperationalLimitDirectionKind.absoluteValue" />
+        <eu:OperationalLimitType.kind rdf:resource="http://iec.ch/TC57/CIM100-European#LimitKind.patl" />
+    </cim:OperationalLimitType>
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml]: You need only one PATL `OperationalLimitType`. You can remove this one.

File: cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        <cim:ActivePowerLimit.normalValue>109</cim:ActivePowerLimit.normalValue>
+        <cim:OperationalLimit.OperationalLimitSet rdf:resource="#_ACLineSegment-2-T1-OperationalLimitSet"/>
+        <cim:OperationalLimit.OperationalLimitType rdf:resource="#_ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit2-Tatl"/>
+    </cim:ActivePowerLimit>
+    <cim:OperationalLimitType rdf:ID="_ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit2-Tatl">
+        <cim:IdentifiedObject.name>ACLineSegment-2-T1-OperationalLimitSet-ActivePowerLimit2-Tatl</cim:IdentifiedObject.name>
+        <cim:OperationalLimitType.acceptableDuration>901</cim:OperationalLimitType.acceptableDuration>
+        <cim:OperationalLimitType.direction rdf:resource="http://iec.ch/TC57/CIM100#OperationalLimitDirectionKind.absoluteValue" />
+        <eu:OperationalLimitType.kind rdf:resource="http://iec.ch/TC57/CIM100-European#LimitKind.tatl" />
+    </cim:OperationalLimitType>
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml]: Strictly speaking this one has a duration of 901s so it's unique, but you can reuse the one with a duration of 900s and remove this `OperationalLimitType`.

If you accept this change, you would also have to adapt the unit tests and in particular make the tatl duration a fixed value (no need to pass it to `ActivePowerLimit` record).

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+        assertEquals(1, network.getTieLineCount());
+
+        TieLine tieLine = network.getTieLine("ACLineSegment-1 + ACLineSegment-2");
+        assertFlow(tieLine.getDanglingLine1(), 275.1, 50.5);
+        assertFlow(tieLine.getDanglingLine2(), -275.0, -50.0);
+        assertBusVoltage(tieLine.getDanglingLine1().getTerminal().getBusView().getBus(), 400.5, -3.0);
+        assertBusVoltage(tieLine.getDanglingLine2().getTerminal().getBusView().getBus(), 402.5, -5.0);
+        assertBoundaryBusVoltage(tieLine.getDanglingLine1(), 401.5130326083143, -4.023034681728034);
+        assertBoundaryBusVoltage(tieLine.getDanglingLine2(), 401.5130326083143, -4.023034681728034);
```
Comment (rcourtier) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java]: You could factorize these assertions with the exact same ones in the following test `importSvSeparatelyTest`.

**Review by rcourtier — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml (hunk 1 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml]: Done

File: cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml (hunk 2 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/test/resources/update/tie-line/tieLine_EQ.xml]: Done

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java (hunk 3 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 4]
```diff
@@ -706,76 +747,21 @@ private void convertEquipmentAtBoundaryNode(Context context, String node) {
...
-                context.fixed(node, "More than two connected AcLineSegments at boundary: only dangling lines are created." +
-                        " Please note that the converted IIDM network will probably not be equivalent to the CGMES network.");
-                beqs.forEach(beq -> beq.createConversion(context).convertAtBoundary());
-            }
+        } else {
+            // In some TYNDPs, there are three or more AcLineSegments at the boundary node, but only two are connected.
+            // Here, a danglingLine is created for each segment, and later, in the method
+            // createTieLinesWhenThereAreMoreThanTwoDanglingLinesAtBoundaryNodeDuringUpdate,
+            // a tieLine is created using only the two connected danglingLines
+            context.fixed(node, "More than two connected AcLineSegments at boundary: only dangling lines are created." +
```
Comment (olperr1) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java]: Here, we don't know yet if the AcLineSegments are connected or not:

```suggestion
            context.fixed(node, "More than two AcLineSegments at boundary: only dangling lines are created." +
```

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 5]
```diff
@@ -269,9 +264,75 @@ public Network convert(ReportNode reportNode) {
...
+    private void updateWithAllInputs(Network network, ReportNode reportNode) {
+        if (!sshOrSvIsIncludedInCgmesModel(this.cgmes)) {
+            return;
+        }
+        this.cgmes.setQueryCatalog(QUERY_CATALOG_NAME_UPDATE);
+        Context updateContext = createUpdateContext(network, reportNode);
+
+        // add processes to create new equipment using update data (ssh and sv data)
+        createFictitiousSwitchesForDisconnectedTerminalsDuringUpdate(network, cgmes, updateContext);
+        createTieLinesWhenThereAreMoreThanTwoDanglingLinesAtBoundaryNodeDuringUpdate(network, updateContext);
```
Comment (olperr1) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:282]: The PR description states:

> When the EQ file and the SSH file are received separately, they are never created and are considered unpaired danglingLines

I think this is caused by the fact that here the switches are not yet updated.
But did something prevent to call this method in the `update(...)`, after the switches and the dangling lines are updated?

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+import org.junit.jupiter.api.Test;
+
+import static com.powsybl.cgmes.conversion.test.ConversionUtil.readCgmesResources;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Luma Zamarreño {@literal <zamarrenolm at aia.es>}
+ * @author José Antonio Marqués {@literal <marquesja at aia.es>}
+ */
+class TieLineUpdateTest {
```
Comment (olperr1) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java:21]: Can you add tests with 3 dangling lines at the same boundary node to test the behavior when importing?
- EQ + EQ_BD + SSH, all at once
- EQ + EQ_BD, then SSH

Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/TieLineUpdateTest.java:21]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java (hunk 4 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java (hunk 5 above)
Comment (marqueslanauja) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:282]: > The PR description states:
> 
> > When the EQ file and the SSH file are received separately, they are never created and are considered unpaired danglingLines
> 
> I think this is caused by the fact that here the switches are not yet updated. But did something prevent to call this method in the `update(...)`, after the switches and the dangling lines are updated?

When there are more than two ACLineSegments connected to a boundary node, one DanglingLine is created for each ACLineSegment. If the EQ and SSH files are imported together, a TieLine is created when there are exactly two ACLineSegments connected to the boundary node. This is done in the `updateWithAllInputs` method, after the conversion process and before the update. However, if EQ and SSH are imported separately, no TieLines are created in this configuration, because the connected status is only available in the SSH file and no equipment is created during the update process.

We tried to preserve the behavior of the current version (EQ + SSH in one shot) as much as possible
